### PR TITLE
Registration

### PR DESCRIPTION
**Pull request by Topikana**
Add new form registration. 
Mail sending to confirm registration user.

### CODE REVIEW

File: src/AppBundle/Controller/RegistrationController.php [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            $company->setTown($form['company']['town']->getData());
+            $company->setPhone2($form['company']['phone2']->getData());
+            $company->setSiret($form['company']['siret']->getData());
+            $company->setEmail($form['company']['email']->getData());
+
+            $userProfile->setFirstname($form['userProfile']['firstname']->getData());
+            $userProfile->setLastname($form['userProfile']['lastname']->getData());
+            $userProfile->setPhone1($form['userProfile']['phone1']->getData());
+
+            $em = $this->getDoctrine()->getManager();
```
Comment (2binfree) [inline src/AppBundle/Controller/RegistrationController.php]: I'm not sure that your code works correctly. Where do you create the links of entities?

File: src/AppBundle/Controller/RegistrationController.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                    ->setSubject('CommunIt : Confirmation de inscription')
+                    ->setFrom('f.letellier0@gmail.com')
+                    ->setTo($user->getEmail())
+                    ->setBody(
+                        $this->renderView('@App/Email/confirm.html.twig', [
+                            'name' => $user->getUserProfile()->getFirstname(),
+                        ]),
+                            'text/html'
+                    );
+                $this->get('mailer')->send($email);
```
Comment (2binfree) [inline src/AppBundle/Controller/RegistrationController.php]: are you sure that your mail is really sent ?

File: src/AppBundle/Form/UserType.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+class UserType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('email', EmailType::class,  ['label' => 'email'])
+            ->add('plainPassword', RepeatedType::class, array(
+                'type' => PasswordType::class,
+                'first_options' => array('label' => 'Mot de passe'),
+                'second_options' => array('label' => 'Répéter votre mot de passe'),
```
Comment (2binfree) [inline src/AppBundle/Form/UserType.php]: Not very elegant formulation

File: src/AppBundle/Resources/views/Email/confirm.html.twig [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        </div>
+    </div>
+</header>
+<div class="main" style="font-size: 16px;padding-left:3rem;">
+    <p>
+        Bonjour {{ name }},
+    </p>
+    <p>
+        Nous vous remercions pour votre pré-inscription sur CommunIt. <br />
+        Votre inscription sera validé dans un délai de 24 heures par l'un de nos administrateurs.<br/>.
```
Comment (2binfree) [inline src/AppBundle/Resources/views/Email/confirm.html.twig]: Faults to be corrected

File: src/AppBundle/Controller/RegistrationController.php (hunk 1 above)
Comment (Topikana) [inline src/AppBundle/Controller/RegistrationController.php]: It is working. The links between are making in the Form/UserType. It's not beautiful...

File: src/AppBundle/Controller/RegistrationController.php (hunk 2 above)
Comment (Topikana) [inline src/AppBundle/Controller/RegistrationController.php]: Yes sure it's working ! i try with differents email adress.

File: src/AppBundle/Form/UserType.php (hunk 3 above)
Comment (Topikana) [inline src/AppBundle/Form/UserType.php]: i try "Confirmation mot de passe" ?

File: src/AppBundle/Resources/views/Email/confirm.html.twig (hunk 4 above)
Comment (Topikana) [inline src/AppBundle/Resources/views/Email/confirm.html.twig]: Nous vous remercions pour votre préinscription sur CommunIt.  Votre inscription sera validée dans un délai de 24 heures par l'un de nos administrateurs.

**Pull request closed without merging**